src/timer.h

```diff
@@ -54,69 +59,79 @@ typedef struct timespec hr_timer;
-#define HRSETCURRENT(X)       gettimeofday(&(X), NULL)
-#define HRGETTICKS(X)         ((uint64_t)(X).tv_sec * 1000000 +	\
-                               (uint64_t)(X).tv_usec)
-#define HRGETTICKS_PER_SEC(X) (X) = 1000000
+
+#define HRZERO(X)         (X).tv_sec = (X).tv_usec = 0
+#define HRSETCURRENT(X)   gettimeofday(&(X), NULL)
+#define HRGETTICKS(X)     ((uint64_t)(X).tv_sec * 1000000 + (uint64_t)(X).tv_usec)
+#define HRSETRESOLUTION() (timer_ticks_per_sec = 1000000)
+#define HRTICKS_PER_SEC   (1000000)
+#define timer_get_nano()  ({ hr_timer t; HRSETCURRENT(t); HRGETTICKS(t) * 1000; })
 
 #endif
 
-#endif /* Windows or Unix */
-
-typedef struct sTimer_s {
-	int m_fRunning;     // true if we are running
-	clock_t m_cStartTime;
-	clock_t m_cEndTime;
-	hr_timer m_hrStartTime;
-	hr_timer m_hrEndTime;
-	double m_dAccumSeconds;
-} sTimer;
-
-extern void sTimer_Init(sTimer *t);      // Init
-extern void sTimer_Start(sTimer *t, int bClear /*=true*/);  // Start the timer
-extern void sTimer_Stop(sTimer *t);      // Stop the timer
-extern void sTimer_ClearTime(sTimer *t); // Clears out the time to 0
-extern double sTimer_GetSecs(sTimer *t); // If timer running returns elapsed;
-                                         // if stopped returns timed interval;
-                                         // if not started returns 0.
-
-extern uint64_t sm_HRTicksPerSec;  // HR ticks per second (claimed)
-extern uint64_t sm_hrPrecision;    // HR ticks per second (observed, guess)
-extern uint64_t sm_cPrecision;     // clocks (ticks) per second (observed, guess)
-
-#define sTimer_Start_noclear(t) \
+#endif /* defined (_MSC_VER) || defined (__MINGW32__) || defined (__CYGWIN32__) */
+
+typedef struct hr_timer_s {
+	int running;
+	hr_timer start;
+	hr_timer end;
+	uint64_t accumulated;
+} hr_timer_t;
+
+extern void timer_init(hr_timer_t *t);             // Init
+extern void timer_start(hr_timer_t *t, int clear); // Start the timer
+extern void timer_stop(hr_timer_t *t);             // Stop the timer
+extern void timer_clear(hr_timer_t *t);            // Clears out the time to 0
+extern double timer_get_secs(hr_timer_t *t);       // Get current time, in seconds
+// uint64_t timer_get_nano()                       // Get current time, in ns (macro)
+extern uint64_t timer_stats(int *latency);         // Return precision/latency
+
+extern uint64_t timer_ticks_per_sec;               // Really needed for Windows & macOS; Set just once
+
+#define timer_start_noclear(t) \
     do { \
-    if (sm_HRTicksPerSec) { HRSETCURRENT ((t)->m_hrStartTime); } \
-    else { (t)->m_cStartTime = clock(); } \
-    (t)->m_fRunning = 1; \
+    HRSETCURRENT ((t)->start); \
+    (t)->running = 1; \
     } while (0)
 
-#define sTimer_Pause(t) \
+#define timer_pause(t) \
     do { \
-    (t)->m_dAccumSeconds = sTimer_GetSecs(t); \
-    (t)->m_fRunning=0; \
+    (t)->accumulated = timer_get_nano(t); \
+    (t)->running=0; \
     } while (0)
```

Comment:
We're not making any use of `timer_pause` yet, so maybe we should drop it along with the `accumulated` field and its updates by other functions/macros.

BTW, there are inconsistently lacking spaces around the `=` in `(t)->running=0;`